sql/connect/common/src/main/protobuf/spark/connect/pipelines.proto

```diff
@@ -90,6 +92,27 @@ message PipelineCommand {
     optional string format = 8;
   }
 
+  // Metadata about why a query function failed to be executed successfully.
+  message QueryFunctionFailure {
+    oneof reason {
+      // The logical plan that the query function needed to eagerly analyze in order to know
+      // the schema / isStreaming / etc of the plan it produced, but could not because it has
+      // unresolved dependencies.
+      spark.connect.Relation unresolved_dependency_plan = 1;
```

Comment:
Vs. passing a list of missing dependencies, which will lead to a simpler backend implementation?